### DISCUSSION TITLE
FO: improve Stock Alert

### DISCUSSION THREAD

**Discussion by Murat-06** [PrestaShop/PrestaShop#32439] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

before I shared an information on this topic. Out of stock warning when the last product in stock is added to the cart

### Expected behavior

Since I sell old products, I usually only sell 1 piece of product. When my customers enter my site, they add the product to their cart and leave, the product is out of stock.
![229995124-66497251-78d0-43c4-9ede-afd29519f211](https://user-images.githubusercontent.com/105777131/232034935-be1727ba-9337-43c6-9551-e6ea6d6f913c.png)
As seen in the picture, the product has 1 in stock.
![229995595-6bae59e1-4147-4a41-975f-b9e937138d6f](https://user-images.githubusercontent.com/105777131/232036115-8e66cfc9-b299-409d-9369-99e0915b190e.png)
Product Added to Cart but out of stock message
When the product is added to the cart, it gives an out of stock message.
The customer does not remember adding the product to the cart and when she comes back, she does not remember that she added it to the cart and leaves the site because she sees the out of stock warning.
suggestion and solution should be like the picture below
![229997004-fe891838-439a-4b8b-b15d-6f6dd7a085b1](https://user-images.githubusercontent.com/105777131/232036607-a3e93ee0-24be-46f4-8334-5f1a129f07d0.png)

### Steps to reproduce

When the product is added to the cart, the product is added to the cart message and the product added to the cart does not fall out of the stock

### PrestaShop version(s) where the bug happened

 kind of versions

### PHP version(s) where the bug happened

 kind of versions

### If your bug is related to a module, specify its name and its version

no

### Your company or customer's name goes here (if applicable).

Murat-06

**Comment by joecoolio66** [comment]
Prestashop has had this issue for a long time... we get occasional emails from customers wondering if the item in their cart is even available when the product page shows out of stock. The stock management kicks in when an item is added to the cart, but it should kick in when the purchase is made.  Multiple people should be able to add the same last item to a cart.

**Comment by hibatallahAouadni** [comment]
Hello @Murat-06 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by Murat-06** [comment]
This stock warning I mentioned is a serious problem, especially when the customer adds the product to the cart and returns without placing an order, and when he enters the product, the out of stock warning causes the customer to leave without placing the order. It's ridiculous that when there is a product in stock, it is added to the cart and it appears out of stock. It should say "Added to the cart" instead of out of stock.

**Reply by hogo20** [reply]
@Murat-06
According to you, someone can easily break up the entire PrestaShop store by adding a product to shopping cart and abandoning the site.
🤔

**Reply by Murat-06** [reply]
I think you responded to the above example images by pointing out that your answer has nothing to do with my question. I have examined all professional e-commerce sites. When you add a product that has 1 item left in stock to the cart, it does not warn that the stock status is out of stock, it gives a warning that it has been added to the cart.

**Reply by kpodemski** [reply]
I agree @Murat-06 - this behavior is a bit unfortunate.

**Reply by hogo20** [reply]
@Murat-06
Hi. Good time.
Yes, I misunderstood. 
 I think if you can change the translation, so-so your problem will be solved.
For example, instead of "Out of stock", change it to "This item is in your cart or Out of stock ." 
In this case, once again, the customer checks their cart!

**Reply by Murat-06** [reply]
Thank you for your suggestion, that's how I did it, I added a check your cart warning. However, the normal working system should not be like this. Professional e-commerce sites give warnings such as "Added to Your Cart", "The product is available in your cart". It would be good if Prestashop developers pay attention to this area. Also, it is not normal for a product that has not been sold to be out of stock. In Prestashop 1.6 version, the product added to the cart is not treated as if it is out of stock.

**Reply by MatShir** [reply]
A PR is incoming https://github.com/PrestaShop/PrestaShop/pull/33931 ;)
